### DISCUSSION TITLE
[V3] Query params bug on empty input

### DISCUSSION THREAD

**Discussion by ariaieboy** [livewire/livewire#5932] [7. Bugs (archived)]
according to this [issue ](https://github.com/filamentphp/filament/issues/7129#issuecomment-1647849202) livewire have 2 bug in the query params section.
first, it changes the empty input to `null` when the history option is set to false.
second, it's not removing the empty query params while the keep option is set to false.
for example, create a component like this:
```php
class Test extends Component
{
    #[Url(history: false,keep: false)]
    public string $search;

    public function render()
    {
        return <<<'HTML'
        <div>
            <input type="text" wire:model.live="search">
        </div>
        HTML;
    }
}
```
now create a route that point to this component for example `localhost/test`
if you hit the URL `localhost/test?search=` it will change the URL to `localhost/test?search=null` while it should change the URL to `localhost/test` since we said don't keep the default value.
if you change the history option to true it will not change the URL to `localhost/test?seach=null` and the URL will stay `localhost/test?search=` but it's still not good It might change the URL to `localhost/test` since the `keep` option is set to false.

**Comment by joshhanley** [comment]
@ariaieboy thanks for reporting! Any chance you could submit separate failing test PRs for each of the bugs so we can investigate it? Thanks!

**Reply by ariaieboy** [reply]
I will try to add a failing test for this, but Browser testing is outside my comfort zone.

**Reply by joshhanley** [reply]
@ariaieboy thanks, it's much appreciated! I believe in you! 🙂 Give it a go, and if you need any guidance, I'm happy to help. Have a look at some of the existing browser tests to get an idea of what to do. They're pretty straight forward once you get the hang of it.

Have a look at this test, to see how we are structuring the tests now https://github.com/livewire/livewire/blob/main/src/Features/SupportMorphAwareIfStatement/BrowserTest.php

And have a look at this test for how to use Dusk to test query strings https://github.com/livewire/livewire/blob/main/legacy_tests/Browser/QueryString/Test.php

For your tests, can you put them in `src/Features/SupportQueryString/BrowserTest.php` (which doesn't yet exist).

Thanks!

**Reply by ariaieboy** [reply]
My problem is that dusk not working properly on windows 😔
It says 127.0.0.1 connection refuse

**Reply by joshhanley** [reply]
@ariaieboy ah that's a pain. No worries, if you can just write the tests the best you can and submit it, then I will make sure they run properly 🙂

**Reply by ariaieboy** [reply]
It's 3:45 here so my brain is not working properly. I created a failing test here https://github.com/livewire/livewire/pull/5943 I hope it helps

**Reply by joshhanley** [reply]
@ariaieboy thanks, much appreciated! Looks good 🙂
